pinot-core/src/main/java/com/linkedin/pinot/core/operator/transform/TransformExpressionOperator.java

```diff
@@ -0,0 +1,161 @@
+ */
+package com.linkedin.pinot.core.operator.transform;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.request.transform.TransformExpressionTree;
+import com.linkedin.pinot.core.common.Block;
+import com.linkedin.pinot.core.common.BlockDocIdIterator;
+import com.linkedin.pinot.core.common.BlockDocIdSet;
+import com.linkedin.pinot.core.common.BlockId;
+import com.linkedin.pinot.core.common.Constants;
+import com.linkedin.pinot.core.indexsegment.IndexSegment;
+import com.linkedin.pinot.core.operator.BaseOperator;
+import com.linkedin.pinot.core.operator.MProjectionOperator;
+import com.linkedin.pinot.core.operator.blocks.IntermediateResultsBlock;
+import com.linkedin.pinot.core.operator.blocks.ProjectionBlock;
+import com.linkedin.pinot.core.operator.docidsets.DocIdSetBlock;
+import com.linkedin.pinot.core.plan.DocIdSetPlanNode;
+
+
+/**
+ * Class for evaluating transform expressions.
+ */
+public class TransformExpressionOperator extends BaseOperator {
+  private static final String OPERATOR_NAME = "TransformExpressionOperator";
+
+  private final IndexSegment _indexSegment;
+  private final MProjectionOperator _projectionOperator;
+  TransformExpressionEvaluator _expressionEvaluator;
+  private int _nextBlockCallCounter = 0;
+
+  // Reusable thread-local int array to store the doc ids.
+  private static final ThreadLocal<int[]> REUSABLE_DOC_ID_SET = new ThreadLocal<int[]>() {
+    @Override
+    protected int[] initialValue() {
+      return new int[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    }
+  };
+
+  /**
+   * Constructor for the class
+   *
+   * @param indexSegment Segment to process
+   * @param projectionOperator Projection operator
+   * @param expressionTree Expression tree to evaluate
+   */
+  public TransformExpressionOperator(IndexSegment indexSegment, MProjectionOperator projectionOperator,
+      TransformExpressionTree expressionTree) {
+
+    Preconditions.checkArgument((indexSegment != null));
+    Preconditions.checkArgument((projectionOperator != null));
+    Preconditions.checkArgument((expressionTree != null));
+
+    _indexSegment = indexSegment;
+    _projectionOperator = projectionOperator;
+
+    _expressionEvaluator = new DefaultExpressionEvaluator(indexSegment, expressionTree);
+  }
+
+  @Override
+  public Block getNextBlock() {
+    return getNextBlock(new BlockId(_nextBlockCallCounter++));
+  }
+
+  @Override
+  public Block getNextBlock(BlockId blockId) {
+    // Only one block supported, currently.
+    if (blockId.getId() > 0) {
+      return null;
+    }
+
+    int numDocsScanned = 0;
+    final long startTimeMillis = System.currentTimeMillis();
+
+    _expressionEvaluator.init();
+    while (_projectionOperator.nextBlock() != null) {
+      ProjectionBlock currentBlock = _projectionOperator.getCurrentBlock();
+      Block docIdSetBlock = currentBlock.getDocIdSetBlock();
+      numDocsScanned = processBlock(currentBlock, docIdSetBlock, numDocsScanned);
```

Comment:
No need to pass in docIdSetBlock because whether it can be used depends on the class of currentBlock.
Also, no need to pass in numDocsScanned, just return the docsScanned for the currentBlock, and add it to the numDocsScanned.